src/new/wallet.js

```diff
@@ -854,7 +857,7 @@ class HathorWallet extends EventEmitter {
     const options = Object.assign({ pinCode: null, password: null }, optionsParams);
     const pinCode = options.pinCode || this.pinCode;
     const password = options.password || this.password;
-    if (!pinCode) {
+    if (!this.xpub && !pinCode) {
```

Comment:
There are more methods in this wallet class where you must have the xpriv to run, e.g. send tx, create token, mint, ... all those methods should be protected and throw an error if you try to call them and you have only the xpub.

Reply:
Done